vm/src/obj/objfloat.rs

```diff
@@ -575,7 +575,30 @@ impl PyFloat {
             "nan" => Ok(std::f64::NAN),
             "inf" => Ok(std::f64::INFINITY),
             "-inf" => Ok(std::f64::NEG_INFINITY),
-            _ => Err(vm.new_value_error("invalid hexadecimal floating-point string".to_string())),
+            value => {
+                let err_msg = "invalid hexadecimal floating-point string";
+                let res = if value.contains("0x") {
```

Comment:
Instead of assigning `res` first, and then using it as an expression, you could opt to not create a `res` variable and use the `if` as the value.